src/components/Template/Template.tsx

```diff
@@ -0,0 +1,225 @@
+  );
+};
+
+/**
+ * This optional component should be the first child of the `Template`
+ * component. This is rendered as an HTML `<header>` element.
+ */
+const TemplateHeader = (props: React.PropsWithChildren<TemplateProps>) => {
+  const styles = useStyleConfig("TemplateHeader", {});
+  return (
+    <Box as="header" __css={styles}>
+      {props.children}
+    </Box>
+  );
+};
+
+/**
+ * This optional component should be used inside the `TemplateHeader` component.
+ * This is meant to render its children from edge to edge and is most useful
+ * for the `Breadcrumbs` and `Hero` components, and other banner-like components.
+ */
+const TemplateContentBreakout = (
+  props: React.PropsWithChildren<TemplateProps>
+) => {
+  const styles = useStyleConfig("TemplateContentBreakout", {});
+  return <Box __css={styles}>{props.children}</Box>;
+};
+
+/**
+ * This is most useful component to render content on the page. This renders an
+ * HTML `<main>` element. This takes a `sidebar` prop with optional "left" or
+ * "right" values. This will pass down the correct Chakra styling for the
+ * `TemplateContentPrimary` and `TemplateContentSidebar` components. Note that
+ * `TemplateContentPrimary` and `TemplateContentSidebar` must be ordered
+ * correctly for the appropriate styles to take effect.
+ */
+const TemplateContent = (
+  props: React.PropsWithChildren<TemplateContentProps>
+) => {
+  const { sidebar = "none", children } = props;
+  const styles = useStyleConfig("TemplateContent", {
+    variant: sidebar !== "none" ? "sidebar" : null,
+  });
+  // Manually pass in the `sidebar` prop to the `TemplateContentPrimary` and
+  // `TemplateContentSidebar` components.
+  const newChildren = React.Children.map(
+    children,
+    (child: React.ReactElement) => {
+      let newChild = child;
+      if (
+        (child && child?.type === TemplateContentPrimary) ||
+        (child?.props && child.props?.mdxType === "TemplateContentPrimary") ||
+        child?.type === TemplateContentSidebar ||
+        (child?.props && child.props?.mdxType === "TemplateContentSidebar")
+      ) {
+        newChild = React.cloneElement(child, { sidebar });
+      }
+
+      return newChild;
+    }
+  );
+
+  return (
+    <Box as="main" __css={styles}>
+      {newChildren}
+    </Box>
+  );
+};
+
+/**
+ * This optional component must be used inside the `TemplateContent` component.
+ * This renders content in the main width of the container and will always render
+ * above the primary component and the sidebar component (if any).
+ */
+const TemplateContentTop = (props: React.PropsWithChildren<TemplateProps>) => {
+  const styles = useStyleConfig("TemplateContentTop", {});
+  return <Box __css={styles}>{props.children}</Box>;
+};
+
+/**
+ * This component is used to render content in a column when there must be a
+ * sidebar component on either its left or right side. It must go inside the
+ * `TemplateContent` component. An optional `sidebar` prop value of "left" or
```

Comment:
For `TemplateContentPrimary` and `TemplateContentSidebar`, this is implemented as a prop so that the parent can add the value at its own level. So while the user doesn't need to set it when they render it, the parent `TemplateContent` internally passes this value. So yes, this prop value is set at the parent level right now, but it still needs to be declared.